nym-vpnc/src/main.rs

```diff
@@ -95,3 +96,16 @@ async fn import_credential(
 fn parse_encoded_credential_data(raw: &str) -> bs58::decode::Result<Vec<u8>> {
     bs58::decode(raw).into_vec()
 }
+
+async fn listen_to_status(client_type: ClientType) -> Result<()> {
+    let mut client = vpnd_client::get_client(client_type).await?;
+    let request = tonic::Request::new(Empty {});
+    let mut stream = client
+        .listen_to_connection_status(request)
+        .await?
+        .into_inner();
+    while let Some(response) = stream.message().await? {
+        println!("{:?}", response);
+    }
+    Ok(())
```

Comment:
should we perhaps log an additional message (maybe to stderr, or as an `error!` log) when we exit the loop, i.e. the channel closes? My understanding is that we never expect it to shutdown, unless the app itself is going off

Reply:
it's not handling the `Ok(None)` case but doesn't the `?` early return cover the case you are mentioning?